iron-image.html

```diff
@@ -263,53 +267,35 @@
       },
 
       observers: [
-        '_transformChanged(sizing, position)'
+        '_transformChanged(sizing, position)',
+        '_loadStateObserver(src, preventLoad)'
       ],
 
-      ready: function() {
-        var img = this.$.img;
-
-        img.onload = function() {
-          if (this.$.img.src !== this._resolveSrc(this.src)) return;
-
-          this._setLoading(false);
-          this._setLoaded(true);
-          this._setError(false);
-        }.bind(this);
-
-        img.onerror = function() {
-          if (this.$.img.src !== this._resolveSrc(this.src)) return;
-
-          this._reset();
-
-          this._setLoading(false);
-          this._setLoaded(false);
-          this._setError(true);
-        }.bind(this);
-
+      created: function() {
         this._resolvedSrc = '';
       },
 
-      _load: function(src) {
-        if (src) {
-          this.$.img.src = src;
-        } else {
-          this.$.img.removeAttribute('src');
+      _imgOnLoad: function() {
+        if (this.$.img.src !== this._resolveSrc(this.src)) {
+          return;
         }
-        this.$.sizedImgDiv.style.backgroundImage = src ? 'url("' + src + '")' : '';
 
-        this._setLoading(!!src);
-        this._setLoaded(false);
+        this._setLoading(false);
```

Comment:
Yeah, it would definitely be better to update these atomically if possible but I don't think there's a way to do this in 1.x and I'd prefer not to implement it locally.